src/sphinxcontrib/towncrier/__init__.py

```diff
@@ -66,7 +67,7 @@ def _get_changelog_draft_entries(
 
 
 @lru_cache(maxsize=1, typed=True)
-def _get_draft_version_fallback(strategy: str, sphinx_config: Dict[str, Any]):
+def _get_draft_version_fallback(strategy: str, sphinx_config: Config):
```

Comment:
Also annotate the return type:
```suggestion
def _get_draft_version_fallback(strategy: str, sphinx_config: SphinxConfig) -> str:
```